packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -43,14 +43,31 @@ const String minSdkVersion = '19';
 const String targetSdkVersion = '33';
 const String ndkVersion = '23.1.7779620';
 
+
+// Update these when new major versions of Java are supported by Gradle.
+// Supported means Java <-> Gradle support.
+// Source of truth: https://docs.gradle.org/current/userguide/compatibility.html
+const String oneMajorVersionHigherJavaVersion = '20';
+
 // Update this when new versions of Gradle come out including minor versions
 //
 // Supported here means supported by the tooling for
 // flutter analyze --suggestions and does not imply broader flutter support.
-const String _maxKnownAndSupportedGradleVersion = '8.0.2';
+const String maxKnownAndSupportedGradleVersion = '8.2.1';
```

Comment:
maybe this should be bumped to 8.3, per https://docs.gradle.org/current/userguide/compatibility.html#java ? I'll trust your judgment though

Reply:
That makes sense. We don't test it currently , so I put it back to 8.0.2. Planning on upgrading this version soon, so I'll do that in a follow-up PR.